compiler/gentype/GenTypeMain.ml

```diff
@@ -90,74 +90,79 @@ let read_cmt cmt_file =
     Log_.item "Try to clean and rebuild.\n\n";
     assert false
 
+let read_input_cmt is_interface cmt_file =
```

Comment:
can you comment on what has changed here, to facilitate review?
otherwise, if you split into two commits: refactor and changes, that's also easy

Reply:
Thank you for the review :) I split up the commits for clarification.